src/sidebar/components/SidebarPanel.tsx

```diff
@@ -56,28 +60,43 @@ export default function SidebarPanel({
     }
   }, [panelIsActive, onActiveChanged]);
 
-  const closePanel = useCallback(() => {
-    store.toggleSidebarPanel(panelName, false);
-  }, [store, panelName]);
+  const onTransitionEnd = useCallback(
+    (direction: 'in' | 'out') => {
+      setShowPanelContent(direction === 'in');
+
+      if (direction !== 'in') {
+        return;
+      }
+
+      const focusEl = initialFocus?.current as HTMLElement & {
+        disabled?: boolean;
+      };
+
+      if (focusEl && !focusEl.disabled) {
+        focusEl.focus();
+      }
+    },
+    [initialFocus],
+  );
```

Comment:
This was previously handled by the `Dialog` component. Bringing here the same logic.